### ISSUE TITLE
gradient interface 

### ISSUE THREAD

**Issue by yebai** [TuringLang/ThArrays.jl#9]
We have the following example for evaluating gradients at the moment. 

```julia
 using ThArrays

 a = rand(3, 2)
 b = rand(3, 2)

ta = Tensor(a, requires_grad=true)
tb = Tensor(b)
tc = ta^2 + 3ta + sin(tb) - tb

tg = Tensor(ones(3, 2))

backward(tc, tg)

grad(ta) 
```

It would be nice to simplify this a bit, e.g. by creating a `gradient(f, x, ...)` API:

```julia

f(ta, tb) =  ta^2 + 3ta + sin(tb) - tb

a = rand(3, 2); b = rand(3, 2)
 # create Tensors with gradient enabled automatically for inputs, ta and tb
# and return a tuple of gradients, i.e. (grad_ta, grad_tb)
ThArrays.gradient(f, ta, tb)
```

**Comment by KDr2** [comment]
Done.
